index.js

```diff
@@ -40,8 +40,11 @@ class ExecutorRouter extends Executor {
             const options = Object.assign({ ecosystem }, plugin.options); // Add ecosystem to executor options
             const executorPlugin = new ExecutorPlugin(options);
 
-            // Set the default executor
-            if (!this.defaultExecutor && !config.defaultPlugin) {
+            if (defaultPlugin && defaultPlugin === plugin.name) {
```

Comment:
Simply doing a `!this.defaultExecutor` as the first `if` will erroneously catch cases where the user defined a `defaultPlugin` *and* that specified plugin is not the first in the array